test/uk/gov/hmrc/tai/connectors/HttpHandlerSpec.scala

```diff
@@ -46,286 +59,281 @@ class HttpHandlerSpec extends BaseSpec with BeforeAndAfterEach {
 
-      okResponse.status mustBe OK
-      okResponse.json mustBe Json.toJson(userInput)
+        response.status mustBe httpStatus
 
-      createdResponse.status mustBe CREATED
-      createdResponse.json mustBe Json.toJson(userInput)
+      }
     }
 
-    "return Http exception" when {
-      "http response is NOT_FOUND" in {
-        when(http.POST[String, HttpResponse](any(), any(), any())(any(), any(), any(), any()))
-          .thenReturn(Future.successful(HttpResponse(NOT_FOUND)))
+    List(
+      GATEWAY_TIMEOUT,
+      INTERNAL_SERVER_ERROR,
+      SERVICE_UNAVAILABLE
+    ).foreach { httpStatus =>
+      s"return Http exception for $httpStatus response" in {
+
+        server.stubFor(
+          post(anyUrl())
+            .willReturn(aResponse().withStatus(httpStatus).withBody("error response")))
 
-        val result = the[HttpException] thrownBy Await.result(sut.postToApi[String](mockUrl, userInput), 5 seconds)
+        val responseFuture = httpHandler.postToApi(testUrl, userInput)
+        val ex = the[Upstream5xxResponse] thrownBy Await.result(responseFuture, 5 seconds)
+        ex.message must include("error response")
 
-        result.responseCode mustBe NOT_FOUND
       }
+    }
 
-      "http response is GATEWAY_TIMEOUT" in {
-        when(http.POST[String, HttpResponse](any(), any(), any())(any(), any(), any(), any()))
-          .thenReturn(Future.successful(HttpResponse(GATEWAY_TIMEOUT)))
+    "return Http exception for BAD_REQUEST response" in {
+      server.stubFor(
+        post(anyUrl())
+          .willReturn(aResponse().withStatus(BAD_REQUEST)))
+      val result = the[BadRequestException] thrownBy Await
+        .result(httpHandler.postToApi[String](testUrl, userInput), 5 seconds)
 
-        val result = the[HttpException] thrownBy Await.result(sut.postToApi[String](mockUrl, userInput), 5 seconds)
+      result.responseCode mustBe BAD_REQUEST
+    }
+
+    "return Http exception for NOT_FOUND response" in {
+
+      server.stubFor(
+        post(anyUrl())
+          .willReturn(aResponse().withStatus(NOT_FOUND)))
+
+      val result = the[NotFoundException] thrownBy Await
+        .result(httpHandler.postToApi[String](testUrl, userInput), 5 seconds)
+
+      result.responseCode mustBe NOT_FOUND
 
-        result.responseCode mustBe GATEWAY_TIMEOUT
-      }
     }
+
   }
 
   "deleteFromApi" must {
-    "post request to DELETE and return the http response" when {
-      "http DELETE returns OK" in {
+    val userInput = "userInput"
 
-        when(http.DELETE[HttpResponse](any(), any())(any(), any(), any()))
-          .thenReturn(Future.successful(HttpResponse(OK)))
-        val result = Await.result(sut.deleteFromApi(mockUrl), 5 seconds)
-        result.status mustBe OK
-      }
+    List(
+      ACCEPTED,
+      OK,
+      NO_CONTENT
+    ).foreach { httpStatus =>
+      s"return json which is coming from http post call for $httpStatus response" in {
```

Comment:
This is not returning json

Reply:
changed name to  s"return $httpStatus status from http post call for $httpStatus response"